src/Microsoft.OData.Core/Buffers/BufferUtils.cs

```diff
@@ -23,12 +23,23 @@ internal static class BufferUtils
         /// <returns>The initialized buffer</returns>
         public static char[] InitializeBufferIfRequired(char[] buffer)
         {
-            if (buffer == null)
+            return InitializeBufferIfRequired(null, buffer);

```

Comment:
Why  do we need this method? can't we do something like this?
if (buffer == null) {
       return  RentFromBuffer(null, buffer);
}
return buffer;

Reply:
That's avoiding a duplicated code. At later, if we change something, we only change the second method, don't need to change all methods. Thanks.

Reply:
I'm okay with that pattern.

---
In reply to: [265697058](https://github.com/OData/odata.net/pull/1420#discussion_r265697058) [](ancestors = 265697058)